src/sentry/testutils/factories.py

```diff
@@ -2215,3 +2218,29 @@ def create_data_condition_group_action(
         return DataConditionGroupAction.objects.create(
             action=action, condition_group=condition_group, **kwargs
         )
+
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.REGION)
+    def create_alert_rule_detector(
+        alert_rule: AlertRule | None = None,
+        detector: Detector | None = None,
+        **kwargs,
+    ) -> AlertRuleDetector:
+        if alert_rule is None:
+            alert_rule = Factories.create_alert_rule()
+        if detector is None:
+            detector = Factories.create_detector()
+        return AlertRuleDetector.objects.create(alert_rule=alert_rule, detector=detector)
+
+    @staticmethod
+    @assume_test_silo_mode(SiloMode.REGION)
+    def create_alert_rule_workflow(
+        alert_rule: AlertRule | None = None,
+        workflow: Workflow | None = None,
+        **kwargs,
+    ) -> AlertRuleWorkflow:
+        if alert_rule is None:
+            alert_rule = Factories.create_alert_rule()
+        if workflow is None:
+            workflow = Factories.create_workflow()
+        return AlertRuleWorkflow.objects.create(alert_rule=alert_rule, workflow=workflow)
```

Comment:
just because these are for our migration only, let's add these to a migration base test class instead. (if we wanted to keep these around after the migration, this is 💯)